src/libs/DateUtils.js

```diff
@@ -291,6 +291,105 @@ function getDateStringFromISOTimestamp(isoTimestamp) {
     return dateString;
 }
 
+/**
+ * @returns {string} example: 2023-05-16
+ */
+function getThirtyMinutesFromNow() {
+    return moment().add(30, 'minutes').format('YYYY-MM-DD HH:mm:ss');
+}
+
+function getOneHourFromNow() {
+    return moment().add(1, 'hours').format('YYYY-MM-DD HH:mm:ss');
+}
+
+/**
+ * @returns {string} example: 2023-05-16
+ */
+function getEndOfToday() {
+    return moment().endOf('day').format('YYYY-MM-DD HH:mm:ss');
+}
+
+/**
+ * @returns {string} example: 2023-05-16
+ */
+function getEndOfWeekFromNow() {
+    return moment().add(7, 'days').format('YYYY-MM-DD HH:mm:ss');
+}
+
+/**
+ * @param {string} time
+ * @returns {string} example: 2023-05-16
+ * it's kind of 'newer' date
+ */
+function extractDate(time) {
+    return moment(time).format('YYYY-MM-DD');
+}
+
+/**
+ * @param {string} dateTimeString
+ * @returns {string} example: 2023-05-16
+ * it's kind of 'newer' date
+ */
+function extractTime(dateTimeString) {
+    return moment(dateTimeString).format('hh:mm');
+}
+/**
+ * @param {string} dateTimeString
+ * @returns {string} example: 2023-05-16
+ * it's kind of 'newer' date
+ */
+function extractTime12Hour(dateTimeString) {
+    return moment(dateTimeString).format('hh:mm A');
+}
+
+/**
+ * @param {string} dateTimeString
+ * @returns {string} example: 2023-05-16
+ * it's kind of 'newer' date
+ */
+function formatDateTo12Hour(dateTimeString) {
+    if (!dateTimeString) return '';
+    return moment(dateTimeString).format('YYYY-MM-DD hh:mm A');
+}
+
+/**
+ * @param {string} type - one of the values from CONST.CUSTOM_STATUS_TYPES
+ * @returns {string} example: 2023-05-16
+ */
+function getDateBasedFromType(type) {
```

Comment:
Since this is specifically for statuses, maybe a clearer function name like `getDateFromStatusType`. (In any case the `Based` can be dropped).

Reply:
make sense
The main idea was to show that the type doesn't contain a date, but we calculate time based on this type.